[bennyLCK] ip

bennyLCK opened the pull request:
# Bond
> "The only easy day was yesterday."  - Navy SEALs ([More information](https://usmilitary.com/the-only-easy-day-was-yesterday-meaning/)).

Bond makes our day today, and in the future **easier** by  recording things we need to do. It's
- text based
- easy to learn
- ~~pretty~~ **SUPER DUPER *FAST*** to use :running_man:

All you need to do is,
1. download it from [***here***](https://github.com/bennyLCK/ip/releases)
2. double-click it 
3. add your tasks
4. let it be your personal task manger **today**!

And the best part of all, is that it's ***ABSOLUTELY* FREE!**

Features include:

- [ ] Managing tasks
- [ ] Managing deadlines (upcoming)
- [ ] Managing reminders (upcoming)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
        Bond bond = new Bond(System.getProperty("user.home") + "/data/Bond.txt");
        bond.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

bennyLCK committed:
Rename the chatbot, implement Greet and Exit messages

bennyLCK committed:
Implement the echo feature

bennyLCK committed:
Revert "Implement the echo feature"

This reverts commit 3df347f58edf711ff0aabd44035681ef1a2d398b.

bennyLCK committed:
Revert "Revert "Implement the echo feature""

This reverts commit 0c26e8042d9f246a3663bcea820b2e60feba7d41.

bennyLCK committed:
Implement Add, List to Bond

bennyLCK committed:
Implement task tracking marking each as done or not done

bennyLCK committed:
Implement ToDo, Deadline and Event task classes and integrate into Bond/Task classes

bennyLCK committed:
Automate Text UI Testing implemented for existing features

bennyLCK committed:
Implement exception handling, using enum in the BondException class for the finite number of commands I allow

bennyLCK committed:
Implement delete to Bond

bennyLCK committed:
Implement save feature where tasks remaining in a previous session is preserved once Bond is run again

bennyLCK committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Implement save feature where tasks remaining in a previous session is preserved once Bond is run again

bennyLCK committed:
Implement Date and Time recognition and representation

bennyLCK committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Implement Date and Time recognition and representation

bennyLCK committed:
Implement more OOP, not yet complete

bennyLCK committed:
Implement Command classes

bennyLCK committed:
Implement more OOP principles

bennyLCK committed:
Divide classes into packages

The packages are named bond.main/.task/.command

bennyLCK committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

bennyLCK committed:
Add JUnit tests

ParserTest: test parcel.parse()

StorageTest: test storage.parseAndAddTask(arg1, arg2)
arg1: String taskFromFile, arg2: ArrayList<Task> tasks

bennyLCK committed:
Build and release of Bond.jar

bennyLCK committed:
Add header comments for all java files

bennyLCK committed:
Add Javadocs to all classes

bennyLCK committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadocs to all classes
  Add header comments for all java files

bennyLCK committed:
Ensure Coding Standard is there

bennyLCK committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Ensure Coding Standard is there

bennyLCK committed:
Implement Find command in Bond

bennyLCK committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement Find command in Bond

Ryo-Seah commented on src/main/java/bond/task/Task.java:
```diff
@@ -0,0 +1,36 @@
+        this.name = name;
+    }
+
+    public void markAsComplete() {
```
Perhaps for boolean setter methods, name them as setComplete() and setInComplete()

Ryo-Seah commented on src/test/java/bond/main/ParserTest.java:
```diff
@@ -0,0 +1,263 @@
+        } catch (BondException e) {
+            assertEquals("Are you for REAL??? No info for a todo", e.getMessage());
+        }
+    }
```
LGTM very comprehensive test cases

Ryo-Seah commented on src/main/java/bond/main/Parser.java:
```diff
@@ -0,0 +1,285 @@
+
+            return new AddEventCommand(taskName, start, end);
+
+        } else if (components[0].equalsIgnoreCase("list")) {
```
Perhaps if you use switch -cases for tthe different type of user input, instead of for the months it would make more sense

Ryo-Seah commented on src/main/java/bond/main/Ui.java:
```diff
@@ -0,0 +1,195 @@
+        this.showLine();
+        this.newLine();
+    }
+
```
LGTM

Ryo-Seah reviewed (commented):
Overall Looks good to me but the months switch statement was quite funny

bennyLCK committed:
Complete checkstyle for all files

bennyLCK committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Complete checkstyle for all files

bennyLCK committed:
Implement basic gui for Bond

bennyLCK committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Implement basic gui for Bond

bennyLCK committed:
Improve GUI for Bond

bennyLCK committed:
Create new JAR file

bennyLCK committed:
Fix minor comment flaw

bennyLCK committed:
Revert "Fix minor comment flaw"

This reverts commit 7429ac2cbbdaed0ca8cc811c80943e14ba360cea.

bennyLCK committed:
Fix minor flaw in comment

bennyLCK committed:
Clean up codes for v0.2

bennyLCK committed:
Include assert statements

Some important assumptions should hold at various points in the code.

Let's add Assert statements to classes including (Main, MainWindow,
Parser, Storage) and to check for expected states of program
behavior or variable value.

By doing so, as a developer of the code, we will be able to see and
rectify flaws in our programming logic which may cause runtime errors.

In addition, the following changes were also made:

* remove unused arguments to some methods in Ui class, also reflecting
such changes in classes using them

* Modifying classes including (Parser, DeadlineTask, EventTask) to now
handle datetime formats of DeadlineTask, EventTask instances better
while also extracting common checks and methods into Parser class for
better adherence to SLAP

* Some minor optimizations to some methods in Ui class and adding of
Javadocs where needed

bennyLCK committed:
Merge pull request #2 from bennyLCK/branch-A-Assertions

Request to merge branch-A-Assertions into branch-master

bennyLCK committed:
Merge branch 'master' of https://github.com/bennyLCK/ip

bennyLCK committed:
Merge branch 'master' into branch-A-Assertions

bennyLCK committed:
Refactor code to improve quality

There are some instances of poor coding practices displayed in previous
commits which hinder understanding of code, making debugging harder.

Let's,
* Modify classes (AddCommand, AddDeadlineCommand, AddEventCommand,
AddToDoCommand) to extract out common functionalities of the three
subclasses of AddCommand into a method addAndStore() to better
adhere to SLAP and reduce code duplication

* Extract out different sections of the parse() method from the Parser
class and the readAndAddTask() from the Storage class to reduce the
length of the method body, while also adhering to SLAP better by
providing an additional layer of abstraction.

* Re-implement the extracted segments from parse() into methods
(parseDeadline(), parseEvent()) using Java Streams to reduce deep
nesting of code logic

* Create local variables for min/max values of (minutes/hours) for
twenty-four hour and Am/Pm time formats respectively to reduce the use
of magic numbers

* Remove unnecessary Function calls, loops, try-catch blocks to
simplify the code and make it easier for others to understand it

bennyLCK committed:
Merge pull request #3 from bennyLCK/branch-A-CodeQuality

Refactor code to improve quality

bennyLCK committed:
Merge branch 'master' of https://github.com/bennyLCK/ip

bennyLCK committed:
Merge branch 'branch-A-Assertions'

bennyLCK committed:
Set up Continuous Integration

On push or pull requests within fork to master,
*Checkout repository
*Merge commit to master
*Validate Gradle Wrapper
*Setup JDK 11
*Build and check with Gradle

bennyLCK committed:
Fix bug regarding class names

*Changed ToDoTask.java to TodoTask.java
*Changed AddToDoCommand.java to AddTodoCommand.java

bennyLCK committed:
Implement update command

bennyLCK committed:
Merge branch 'branch-C-Update'

bennyLCK committed:
Add AI.md file

* Describes how I use GitHub Co-pilot to aid me in coding.

bennyLCK committed:
Include user guide

bennyLCK committed:
Update README

* Rectifies previous wrong information in user guide
for the find method.

* Additional changes also include more comprehensive test cases
for methods in Parser class for find and update methods.
